apps/challenges/aws_utils.py

```diff
@@ -766,57 +767,110 @@ def restart_workers_signal_callback(sender, instance, field_name, **kwargs):
         else:
             challenge = instance
 
-        response = restart_workers([challenge])
+        serialized_challenge = serializers.serialize("json", [challenge])
+        restart_workers_on_file_change.delay(field_name, serialized_challenge)
 
-        count, failures = response["count"], response["failures"]
 
-        logger.info(
-            "The worker service for challenge {} was restarted, as {} was changed.".format(
-                challenge.pk, field_name
-            )
+@app.task
+def restart_workers_on_file_change(field_name, serialized_challenge):
+    """
+    Restarts the challenge workers, and sends email notifications to the hosts.
+    """
+    for obj in serializers.deserialize("json", serialized_challenge):
+        challenge = obj.object
+    response = restart_workers([challenge])
+
+    count, failures = response["count"], response["failures"]
+
+    logger.info(
+        "The worker service for challenge {} was restarted, as {} was changed.".format(
```

Comment:
Message should be `Worker(s) for challenge {} were restarted as {} was changed.`